File: digitalmaktabapi/Models/LearningMaterial.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace digitalmaktabapi.Models
{
    [Table("LearningMaterial")]
    public class LearningMaterial : Base
    {
        public required string Title { get; set; }
        public required string Description { get; set; }
        public string? FilePath { get; set; }
        public string? FileName { get; set; }
        public string? ContentType { get; set; }

        public required LearningMaterialType LearningMaterialType { get; set; }
        public string? ThumbnailPath { get; set; }
        public required Guid CourseSectionId { get; set; }
        public required CourseSection CourseSection { get; set; }
    }
}

File: digitalmaktabspa/src/models/LearningMaterial.ts
import { Base } from "./Base";
import { CourseSection } from "./CourseSection";
import { LearningMaterialType } from "./LearningMaterialType";

export interface LearningMaterial extends Base {
  title: string;
  description: string;
  filePath?: string;
  file?: File | null;
  fileName?: string;
  contentType?: string;
  learningMaterialType: LearningMaterialType;
  learningMaterialTypeValue?: string;
  thumbnailPath?: string;
  thumbnail?: File | null;
  courseSectionId?: string;
  courseSection?: CourseSection;
}
